internal/aws/cwlogs/cwlog_client.go

```diff
@@ -154,9 +156,17 @@ func (client *Client) CreateStream(logGroup, streamName *string) (token string,
 		client.logger.Debug("cwlog_client: creating stream fail", zap.Error(err))
 		var awsErr awserr.Error
 		if errors.As(err, &awsErr) && awsErr.Code() == cloudwatchlogs.ErrCodeResourceNotFoundException {
-			_, err = client.svc.CreateLogGroup(&cloudwatchlogs.CreateLogGroupInput{
-				LogGroupName: logGroup,
-			})
+			// Create Log Group with tags if they exist and were specified in the config
+			if client.tags != nil && len(client.tags) > 0 {
+				_, err = client.svc.CreateLogGroup(&cloudwatchlogs.CreateLogGroupInput{
+					LogGroupName: logGroup,
+					Tags:         client.tags,
+				})
+			} else {
+				_, err = client.svc.CreateLogGroup(&cloudwatchlogs.CreateLogGroupInput{
+					LogGroupName: logGroup,
+				})
+			}
```

Comment:
```suggestion
			// Create Log Group with tags if they exist and were specified in the config
			_, err = client.svc.CreateLogGroup(&cloudwatchlogs.CreateLogGroupInput{
				LogGroupName: logGroup,
				Tags:         client.tags,
			})
```

I think that the tags can be included here always.  `nil` is the zero value so explicitly including it is no different than not including it when the value is `nil`.  If the value is non-`nil` but zero length then there are no tags to add.

Reply:
I have removed the if statement, but I have also changed the validation function to throw an error if value is non-`nil` but zero length because this was giving me an error saying an empty `{}` cannot be used.